diode-server/netboxdiodeplugin/client.go

```diff
@@ -302,16 +302,16 @@ func (c *Client) GenerateDiff(ctx context.Context, payload GenerateDiffRequest)
 
 	c.logger.Debug("generate diff", "statusCode", resp.StatusCode, "response", string(respBytes))
 
-	// return errors with 4xx status code
-	if resp.StatusCode >= http.StatusBadRequest {
-		return nil, changeset.NewError("generate diff failed", diodeErrors.ErrCodeOpsGenerateDiff, respBytes)
-	}
-
```

Comment:
if it returns respBytes here it hasn't yet been parsed as json, so a 500 will give an invalid changset.Error that fails to write to the database.  changed to return after, which will send back json parsing failures as a "failed to unmarshal" error / internal error.